WordPress/src/test/java/org/wordpress/android/ui/mysite/cards/dynamiccard/DynamicCardsBuilderTest.kt

```diff
@@ -0,0 +1,212 @@
+    id = DYNAMIC_CARD_ID,
+    title = DYNAMIC_CARD_TITLE,
+    remoteFeatureFlag = DYNAMIC_CARD_REMOTE_FEATURE_FLAG,
+    featuredImage = DYNAMIC_CARD_FEATURED_IMAGE,
+    url = "",
+    action = null,
+    order = CardModel.DynamicCardsModel.CardOrder.fromString(DYNAMIC_CARD_ORDER),
+    rows = listOf(DYNAMIC_CARD_ROW_MODEL)
+)
+
+private val DYNAMIC_CARD_MODEL_INVALID_URL = CardModel.DynamicCardsModel.DynamicCardModel(
+    id = DYNAMIC_CARD_ID,
+    title = DYNAMIC_CARD_TITLE,
+    remoteFeatureFlag = DYNAMIC_CARD_REMOTE_FEATURE_FLAG,
+    featuredImage = DYNAMIC_CARD_FEATURED_IMAGE,
+    url = null,
+    action = DYNAMIC_CARD_ACTION,
+    order = CardModel.DynamicCardsModel.CardOrder.fromString(DYNAMIC_CARD_ORDER),
+    rows = listOf(DYNAMIC_CARD_ROW_MODEL)
+)
+
+@ExperimentalCoroutinesApi
+class DynamicCardsBuilderTest : BaseUnitTest() {
+    private lateinit var dynamicCardsBuilder: DynamicCardsBuilder
+
+    @Mock
+    private lateinit var urlUtils: UrlUtilsWrapper
+
+    @Mock
+    private lateinit var deepLinkHandlers: DeepLinkHandlers
+
+    @Before
+    fun setUp() {
+        dynamicCardsBuilder = DynamicCardsBuilder(urlUtils, deepLinkHandlers)
+        whenever(urlUtils.isValidUrlAndHostNotNull(DYNAMIC_CARD_URL)).thenReturn(true)
+    }
+
+    @Test
+    fun testBuild() {
+        val expectedCards = listOf(
+            MySiteCardAndItem.Card.Dynamic(
+                id = DYNAMIC_CARD_ID,
+                rows = listOf(
+                    MySiteCardAndItem.Card.Dynamic.Row(
+                        iconUrl = DYNAMIC_CARD_ROW_ICON,
+                        title = DYNAMIC_CARD_ROW_TITLE,
+                        description = DYNAMIC_CARD_ROW_DESCRIPTION,
+                    )
+                ),
+                title = DYNAMIC_CARD_TITLE,
+                image = DYNAMIC_CARD_FEATURED_IMAGE,
+                action = MySiteCardAndItem.Card.Dynamic.ActionSource.Button(
+                    url = DYNAMIC_CARD_URL,
+                    onCtaClick = mock(),
+                    title = DYNAMIC_CARD_ACTION
+                ),
+                onHideMenuItemClick = mock(),
+            )
+        )
+        val builderParams = MySiteCardAndItemBuilderParams.DynamicCardsBuilderParams(
+            dynamicCards = CardModel.DynamicCardsModel(
+                pages = listOf(DYNAMIC_CARD_MODEL)
+            ),
+            onActionClick = mock(),
+            onMoreMenuClick = mock(),
+            onHideMenuItemClick = mock(),
+        )
+        val dynamicCards = dynamicCardsBuilder.build(builderParams, CardModel.DynamicCardsModel.CardOrder.TOP)
+        assertThat(requireNotNull(dynamicCards).size).isEqualTo(1)
+        assertEquals(expectedCards[0].id, dynamicCards[0].id)
+        assertEquals(expectedCards[0].title, dynamicCards[0].title)
+        assertEquals(expectedCards[0].image, dynamicCards[0].image)
+        assertThat(expectedCards[0].rows.size).isEqualTo(1)
+        assertEquals(expectedCards[0].rows, dynamicCards[0].rows)
```

Comment:
NIT: `assertEquals` can be used everywhere I think, but there is no need to change it.